### PR TITLE
Update upstream changes

### PR DESCRIPTION
**Pull request by iamashwin99**
update changes from PR https://github.com/spack/spack/pull/33969

### CODE REVIEW

File: spack/package.py [hunk 1]
```diff
@@ -77,24 +77,27 @@ class Octopus(AutotoolsPackage, CudaPackage):
         depends_on("fftw@3:+mpi+openmp", when="@8:9")  # FFT library
         depends_on("fftw-api@3:+mpi+openmp", when="@10:")
         depends_on("libvdwxc+mpi", when="+libvdwxc")
+        depends_on("arpack-ng+mpi", when="+arpack")
+        depends_on("elpa+mpi", when="+elpa")
+        depends_on("netcdf-fortran", when="+netcdf")
```
[the remark this answers is not in the document]
Comment (iamashwin99) [reply spack/package.py]: commit [323403e](https://github.com/fangohr/octopus-in-spack/pull/58/commits/323403e54ba4cc94eca96fa8c55b181548380958) should solve this

[the remark this answers is not in the document]
Comment (iamashwin99) [reply spack/package.py]: From the action log, and from `spack info netcdf-fortran`  we see that `netcdf-fortran` doesn't have a mpi variant. Its the netcdf-c that has the mpi variant, would you suggest that we just stick to `depends_on("netcdf-fortran", when="+netcdf")` syntax and rely on `+mpi` being a default in the netcdf-c package, or we explicitly use `depends_on("netcdf-fortran ^netcdf-c+mpi", when="+netcdf")`

[the remark this answers is not in the document]
Comment (fangohr) [reply spack/package.py]: Good point.

Yes, I prefer the explicit version: `depends_on("netcdf-fortran ^netcdf-c+mpi", when="+netcdf")`

**Comment by iamashwin99** [comment]
The PR https://github.com/spack/spack/pull/33969 has been merged, The latest commit now syncs the change.

 Open for review.

**Comment by iamashwin99** [comment]
Yes this PR pulls changes from spack/spack develop branch.
The latest commit bumps the year in the licence string.

**Pull request merged**